worker.go

```diff
@@ -43,17 +44,49 @@ func NewWorkerMetrics() WorkerMetrics {
 	}
 }
 
+// Create an entry in wm for the given metrickey, if it does not already exist.
+// Returns true if the metric entry was created and false otherwise.
+func (wm WorkerMetrics) Upsert(mk MetricKey, tags []string) bool {
```

Comment:
Yeah, that was my suspicion. Pardon me, my Scala is showing! (Actually, this would fail there as well since there no type unions.)